mitiq/zne/scaling/layer_scaling.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Functions for layer-wise unitary folding on supported circuits."""
+from typing import List
+from cirq import Circuit, inverse, Moment
+
+
+def layer_folding(circuit: Circuit, layers_to_fold: List[int]) -> Circuit:
```

Comment:
For consistency with the [folding](https://github.com/unitaryfund/mitiq/blob/c849314bc73c2e7b1b71d312639c5e8417ab2e74/mitiq/zne/scaling/folding.py#L219-L223) methods, and the [identity insertion](https://github.com/unitaryfund/mitiq/blob/c849314bc73c2e7b1b71d312639c5e8417ab2e74/mitiq/zne/scaling/identity_insertion.py#L82) scaling method, what if we separated `layers_to_fold` into two separate variables:

1. `layers_to_fold`: To record layer indices
2. `scale_factors`: to record the number of times to fold

I do like how the current API is very dense in it's input data. Not a clear winner for me here, but I lean slightly towards having consistent API if it doesn't seem like a disadvantage to you.